src/test/mir-opt/box_expr.rs

```diff
@@ -53,7 +53,7 @@ impl Drop for S {
 //         StorageLive(_3);
 //         StorageLive(_4);
 //         _4 = move _1;
-//         _3 = const std::mem::drop::<std::boxed::Box<S>>(move _4) -> [return: bb5, unwind: bb7];
+//         drop(_4) -> [return: bb5, unwind: bb7];
```

Comment:
Does this change actually result in generators being smaller? I was under the impression that the move on the line above was the problem.

Reply:
No, this change does not change the size of generators, but it allows removing the `let` binding inside the `.await` expansion, which *would* reduce the size of generators.

Reply:
This doesn't seem to solve that problem either, there's still the move on the line above that needs to be removed.

Reply:
Yes, good catch, that move will need to be eliminated as well.